src/textual/css/_help_text.py

```diff
@@ -0,0 +1,690 @@
+
+import sys
+from dataclasses import dataclass
+
+from textual.css._help_renderables import Example, Bullet, HelpText
+from textual.css.constants import (
+    VALID_BORDER,
+    VALID_LAYOUT,
+    VALID_EDGE,
+    VALID_ALIGN_HORIZONTAL,
+    VALID_ALIGN_VERTICAL,
+    VALID_STYLE_FLAGS,
+)
+
+if sys.version_info >= (3, 8):
+    from typing import Literal, Iterable
+else:
+    from typing_extensions import Literal
+
+if sys.version_info >= (3, 9):
+    from typing import Annotated
+else:
+    from typing_extensions import Annotated
+
+from textual.css._error_tools import friendly_list
+from textual.css.scalar import SYMBOL_UNIT
+
+StylingContext = Annotated[
+    Literal["inline", "css"],
+    """The type of styling the user was using when the error was encountered.
+Used to give help text specific to the context i.e. we give CSS help if the
+user hit an issue with their CSS, and Python help text when the user has an
+issue with inline styles.""",
+]
+
+
+@dataclass
+class ContextSpecificBullets:
+    """
+    Args:
+        inline (Iterable[Bullet]): Information only relevant to users who are using inline styling.
+        css (Iterable[Bullet]): Information only relevant to users who are using CSS.
+    """
+
+    inline: Iterable[Bullet]
+    css: Iterable[Bullet]
+
+    def get_by_context(self, context: StylingContext) -> list[Bullet]:
+        """Get the information associated with the given context
+
+        Args:
+            context (StylingContext | None): The context to retrieve info for.
+        """
+        if context == "inline":
+            return list(self.inline)
+        else:
+            return list(self.css)
+
+
+def _python_name(property_name: str) -> str:
+    """Convert a CSS property name to the corresponding Python attribute name
+
+    Args:
+        property_name (str): The CSS property name
+
+    Returns:
+        str: The Python attribute name as found on the Styles object
+    """
+    return property_name.replace("-", "_")
+
+
+def _css_name(property_name: str) -> str:
+    """Convert a Python style attribute name to the corresponding CSS property name
+
+    Args:
+        property_name (str): The Python property name
+
+    Returns:
+        str: The CSS property name
+    """
+    return property_name.replace("_", "-")
+
+
+def _contextualize_property_name(
+    property_name: str,
+    context: StylingContext,
+) -> str:
+    """Convert a property name to CSS or inline by replacing
+        '-' with '_' or vice-versa
+
+    Args:
+        property_name (str): The name of the property
+        context (StylingContext): The context the property is being used in.
+
+    Returns:
+        str: The property name converted to the given context.
+    """
+    return _css_name(property_name) if context == "css" else _python_name(property_name)
+
+
+def _spacing_examples(property_name: str) -> ContextSpecificBullets:
+    """Returns examples for spacing properties"""
+    return ContextSpecificBullets(
```

Comment:
Both inline and css will be executed on an error occurring since they aren't lazily evaluated. I thought this was fine since it keeps things readable and it's a very minimal overhead that only appears in error cases.